core/src/main/java/org/sonarsource/sonarlint/core/container/global/MapSettings.java

```diff
@@ -75,24 +83,295 @@ public MapSettings clear() {
     return this;
   }
 
+  /**
```

Comment:
I guess this (quite large) insert comes from moving implementation code from the plugin API to SLCORE, right?

Reply:
Yep, the implementation of `Settings` in SQ was moved to the sonar-plugin-api-impl package. I copied most of it, removing the support of `Encryption` to drop the dependency on commons-codec.